src/sage/groups/perm_gps/permgroup.py

```diff
@@ -1581,6 +1581,93 @@ def smallest_moved_point(self):
+            Group([ (1,2,3) ])
+            sage: T = PermutationGroup(gap_group=libgap.DirectProduct(A,B))
+            sage: T.is_isomorphic(H)
+            True
+            sage: PermutationGroup(PermutationGroup(gap_group=A).gens(),domain=list(S[1])).disjoint_direct_product_decomposition()
+            {{4, 5, 6, 7, 8, 9, 10, 11, 12}}
+            sage: PermutationGroup(PermutationGroup(gap_group=B).gens(),domain=list(S[0])).disjoint_direct_product_decomposition()
+            {{1, 2, 3}}
+
+        Counting the number of connected subgroups::
+
+            sage: # optional -- internet
+            sage: seq = [sum(1 for G in SymmetricGroup(n).conjugacy_classes_subgroups() if len(G.disjoint_direct_product_decomposition()) == 1) for n in range(1,8)];seq
+            [1, 1, 2, 6, 6, 27, 20]
+            sage: oeis(seq)
+            0: A005226: Number of atomic species of degree n; also number of connected permutation groups of degree n.
+        """
+        from sage.combinat.set_partition import SetPartition
+        from sage.sets.disjoint_set import DisjointSet
+        H = self._libgap_()
+        if self.is_trivial():
+            return SetPartition(DisjointSet(self.domain()))
+        if libgap.NrMovedPoints(H) == self.degree() and libgap.IsTransitive(H):
+            return SetPartition([self.domain()])
+        O = libgap.Orbits(H)
+        k = len(O)
+        OrbitMapping = dict()
+        for i in range(k):
+            for x in O[i]:
+                OrbitMapping[x] = i
+        C = libgap.StabChain(H, libgap.Concatenation(O))
+        X = libgap.StrongGeneratorsStabChain(C)
+        P = DisjointSet(k)
+        R = libgap.List([])
+        identity = libgap.Identity(H)
+        for i in range(k-1):
+            libgap.Append(R, O[i])
+            Xp = libgap.List([])
+            while True:
+                try:
+                    if libgap.IsSubset(O[i], C['orbit']):
+                        C = C['stabilizer']
+                    else:
+                        break
+                except ValueError:  #this should catch a GAPError but I don't know how to make it work
+                    break
+            for x in X:
+                xs = libgap.SiftedPermutation(C, x)
+                if xs != identity:
+                    cj = OrbitMapping[libgap.SmallestMovedPoint(libgap.RestrictedPerm(x, R))]
+                    libgap.Add(Xp, xs)
+                    if libgap.RestrictedPerm(xs, O[i+1]) != identity:
+                        P.union(i+1, cj)
+                else:
+                    libgap.Add(Xp, x)
+            X = Xp
+        final_partition = DisjointSet(self.domain())
+        for part in P:
+            grp = [self._domain_from_gap[Integer(x)]
+                   for i in part
+                   for x in O[i]]
+            for i in range(1, len(grp)):
+                final_partition.union(grp[0], grp[i])
+        return SetPartition(final_partition)
```

Comment:
Possibly (but I'm not sure!)
```suggestion
        return SetPartition([[self._domain_from_gap[Integer(x)]
                              for i in part
                              for x in O[i]]
                             for part in P]
                            + [[x] for x in self.fixed_points()])
```
is easier to understand.